talk/oms/sdk/base/win/msdkvideoencoder.cc

```diff
@@ -190,20 +194,70 @@ int MSDKVideoEncoder::InitEncodeOnEncoderThread(const webrtc::VideoCodec* codec_
   m_mfxEncParams.mfx.FrameInfo.Width = MSDK_ALIGN16(codec_settings->width);
 
   m_mfxEncParams.AsyncDepth = 4;
-  m_mfxEncParams.mfx.NumRefFrame = 1;
+  m_mfxEncParams.mfx.NumRefFrame = 2;
+
+  mfxExtCodingOption extendedCodingOptions;
+  MSDK_ZERO_MEMORY(extendedCodingOptions);
+  extendedCodingOptions.Header.BufferId = MFX_EXTBUFF_CODING_OPTION;
+  extendedCodingOptions.Header.BufferSz = sizeof(extendedCodingOptions);
+  // extendedCodingOptions.MaxDecFrameBuffering =
+  // m_mfxEncParams.mfx.NumRefFrame;
+  extendedCodingOptions.AUDelimiter = MFX_CODINGOPTION_OFF;
+  // extendedCodingOptions.RecoveryPointSEI = MFX_CODINGOPTION_OFF;
+  extendedCodingOptions.PicTimingSEI = MFX_CODINGOPTION_OFF;
+  extendedCodingOptions.VuiNalHrdParameters = MFX_CODINGOPTION_OFF;
+  // extendedCodingOptions.VuiVclHrdParameters = MFX_CODINGOPTION_OFF;
+  mfxExtCodingOption2 extendedCodingOptions2;
+  MSDK_ZERO_MEMORY(extendedCodingOptions2);
+  extendedCodingOptions2.Header.BufferId = MFX_EXTBUFF_CODING_OPTION2;
+  extendedCodingOptions2.Header.BufferSz = sizeof(extendedCodingOptions2);
+  extendedCodingOptions2.RepeatPPS = MFX_CODINGOPTION_OFF;
+
+  m_EncExtParams.push_back((mfxExtBuffer*)&extendedCodingOptions);
+  m_EncExtParams.push_back((mfxExtBuffer*)&extendedCodingOptions2);
+
+#if (MFX_VERSION >= 1025)
+  uint32_t timeout = MSDKFactory::Get()->GetMFETimeout();
+  if (timeout) {
+    mfxExtMultiFrameParam multiFrameParam;
+    MSDK_ZERO_MEMORY(multiFrameParam);
+    multiFrameParam.Header.BufferId = MFX_EXTBUFF_MULTI_FRAME_PARAM;
+    multiFrameParam.Header.BufferSz = sizeof(multiFrameParam);
+    multiFrameParam.MFMode = MFX_MF_AUTO;
+    m_EncExtParams.push_back((mfxExtBuffer*)&multiFrameParam);
+
+    mfxExtMultiFrameControl multiFrameControl;
+    MSDK_ZERO_MEMORY(multiFrameControl);
+    multiFrameControl.Header.BufferId = MFX_EXTBUFF_MULTI_FRAME_CONTROL;
+    multiFrameControl.Header.BufferSz = sizeof(multiFrameControl);
+    multiFrameControl.Timeout = timeout;
+    m_EncExtParams.push_back((mfxExtBuffer*)&multiFrameControl);
+  }
+#endif
```

Comment:
I'm not quite familiar with this part. Maybe @HuaChunbo has more knowledge on it.